ovs/lib/scheduledtask.py

```diff
@@ -277,13 +282,50 @@ def gather_scrub_work():
                                                                               vdisk_guids=local_vdisks_to_scrub)
             except Exception as ex:
                 logger.error('Gather Scrub - Storage Router {0:<15} - Scrubbing failed with error:\n - {1}'.format(local_storage_router.ip, ex))
-        all_results = result_set.join(propagate=False)  # Propagate False makes sure all jobs are waited for even when 1 or more jobs fail
+
+        all_results, failed_nodes = manage_running_tasks(result_set, timesleep=60)  # Check every 60 seconds if tasks are still running
+
+        for index, result in enumerate(all_results):
+            if isinstance(result, list):
+                processed_guids.extend(result)
+            else:
+                logger.error('Gather Scrub - Storage Router {0:<15} - Scrubbing failed with error:\n - {1}'.format(storage_router_list[index].ip, result))
+
+        result_set = {}
+        for failed_node in failed_nodes:
+            logger.warning('Scrubbing failed on node {0}. Will reschedule on another node.'.format(failed_node))
+            vdisk_guids_to_scrub = scrub_map[failed_node]
+            rescheduled_work = False
+            for storage_router, scrub_location in scrub_locations.items():
+                if storage_router.ip != failed_node:
```

Comment:
If you have multiple failed nodes, this code can potentially start scrubbing on a different failed node, than the 1 being checked currently